tmt/steps/provision/__init__.py

```diff
@@ -923,6 +923,18 @@ def setup(self) -> None:
 
         Setup the guest after it has been started. It is called after :py:meth:`Guest.start`.
         """
+        from tmt.steps.execute.internal import effective_pidfile_root
+        sudo = 'sudo' if not self.facts.is_superuser else ''
+        pid_directory = effective_pidfile_root()
+        self.execute(ShellScript(
+            f"""
+            if [ ! -d {pid_directory} ]; then \
+                   {sudo} mkdir -p {pid_directory} \
+                && {sudo} setfacl -d -m u::rwX,g::rwX,o::rwX \
```

Comment:
The directory is missing in here, right?

@psss @happz @mcasquer,

But also, this logic in general won't work for a system with `umask 0027`, if you want subsequent files and directories in the created directories to be world-writable.

The logic should be something like this:
1. Find out if the `work_dir` directory is prefix of `pid_directory`
2. Make `work_dir` world writable if it is not
3. Set the `o` acl to `rwX` in `work_dir`
4. Create the rest of directory with `mkdir -p {pid_directory}`.


If `work_dir` is not a prefix of `pid_directory`, then you would still need to create `pid_directory` and do chmod recursively, but keep into account that anything else created aftewards in that structure won't be world-writable. Only your pid path.

Here is a test:

Fail scenario:
```sh
# $ podman run -it quay.io/fedora/fedora:latest

dnf install -y /usr/bin/setfacl

umask 0027 # by default, files and directory will be created with no permission for others, even if you do chmod +rwx

mkdir -p /var/tmp/tmt/pid # create the structure
chmod -R o+rwX /var/tmp/tmt # even make it all world writable underneath the work directory
ls -lash /var/tmp/tmt # verify it
ls -lash /var/tmp/tmt/pid # verify it
touch /var/tmp/tmt/pid/pid.lock # however, the file is not, because umask doesn't allow it.
ls -lash /var/tmp/tmt/pid/pid.lock # check it.

# In fact, other directory you create `/var/tmp/tmt` won't have the world-writable permissions:
mkdir /var/tmp/tmt/test
ls -lash /var/tmp/tmt

# A test users:
useradd test
sudo -u test touch /var/tmp/tmt/pid/pid.lock
touch: cannot touch '/var/tmp/tmt/pid/pid.lock': Permission denied
sudo -u test ls /var/tmp/tmt/test
ls: cannot open directory '/var/tmp/tmt/test': Permission denied
```

Success scenario:

```sh
# $ podman run -it quay.io/fedora/fedora:latest

dnf install -y /usr/bin/setfacl
umask 0027

mkdir /var/tmp/tmt
chmod o+rwX /var/tmp/tmt
setfacl -d -m o:rwX /var/tmp/tmt
# Now everything that it is created, directories or files, within /var/tmp/tmt and its subdirectories are created world-writable

mkdir -p /var/tmp/tmt/pid
touch /var/tmp/tmt/pid/pid.lock
s -lash /var/tmp/tmt
# total 0
# 0 drwxr-xrwx+ 1 root root  6 Dec 10 14:41 .
# 0 drwxrwxrwt. 1 root root  6 Dec 10 14:40 ..
# 0 drwxr-xrwx+ 1 root root 16 Dec 10 14:41 pid

ls -lash /var/tmp/tmt/pid
# total 0
# 0 drwxr-xrwx+ 1 root root 16 Dec 10 14:41 .
# 0 drwxr-xrwx+ 1 root root  6 Dec 10 14:41 ..
# 0 -rw-r--rw-. 1 root root  0 Dec 10 14:41 pid.lock

sudo -u test touch /var/tmp/tmt/pid/pid.lock # exit 0
mkdir -p /var/tmp/tmt/test
sudo -u test ls /var/tmp/tmt/test # exit 0
```

Reply:
Assuming it uses the `-p` flag no error is reported and the `workdir_root` directory is created during this process